doc/source/whatsnew/v0.21.0.txt

```diff
@@ -109,6 +109,40 @@ For example:
     # the following is now equivalent
     df.drop(columns=['B', 'C'])
 
+.. _whatsnew_0210.enhancements.rename_reindex_axis:
+
+``rename``, ``reindex`` now also accept axis keyword
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+The :meth:`DataFrame.rename` and :meth:`DataFrame.reindex` methods have gained
+the ``axis`` keyword to specify the axis to target with the operation. This
+harmonizes the method signature of these methods with most of the rest of the
+library (:issue:`12392`).
+
+Here's ``rename``:
+
+.. ipython:: python
+
+   df = pd.DataFrame({"A": [1, 2, 3], "B": [4, 5, 6]})
+   df.rename(str.lower, axis='columns')
+   df.rename(id, axis='index')
+
+And ``reindex``:
+
+.. ipython:: python
+
+   df.reindex(['A', 'B', 'C'], axis='columns')
+   df.reindex([0, 1, 3], axis='index')
+
+The ``.rename(index=id, columns=str.lower)`` style continues to work as before.
+
+.. ipython:: python
+
+   df.rename(index=id, columns=str.lower)
+
```

Comment:
maybe add `df.reindex(index=[0, 1, 3], columns=['A', 'B', 'C'])` as well? to make it clear that it is not only `rename` but also `reindex` that keeps the old way